File: Shared/Interfaces/IStoreProductsService.cs
using GroceryListHelper.Shared.Models.StoreProducts;

namespace GroceryListHelper.Shared.Interfaces;

public interface IStoreProductsService
{
    Task ClearStoreProducts();
    Task<List<StoreProduct>> GetStoreProducts();
    Task SaveStoreProduct(StoreProduct product);
    Task UpdateStoreProduct(StoreProduct storeProduct);
}

File: GroceryListHelper.Client/src/types/IStoreProductsService.ts
import type { StoreProduct } from "./StoreProduct";

export interface IStoreProductsService {
    getStoreProducts: () => Promise<StoreProduct[] | Error>;
    createStoreProduct: (product: StoreProduct) => Promise<Error | null>;
    updateStoreProduct: (product: StoreProduct) => Promise<Error | null>;
    deleteStoreProducts: () => Promise<Error | null>;
}